docs/src/content/docs/sitemap-ext.md

```diff
@@ -0,0 +1,191 @@
+
+sitemap(true); // opt-in
+
+sitemap(true); // opt-out
+---
+```
+
+In this case all pages created from the route will follow the given configuration.
+
+:::caution
+This applies for all pre-rendered routes and to on-demand _static_ routes (routes that don't have dynamic parameters in their path).
+
+On-demand dynamic routes, like `/[...slug].astro`, can't use the full opt-in since the official sitemap integration that is used under the hood must generate the sitemap statically.
+:::
+
+### Configuration callback
+
+For non-trivial scenarios, you can provide a hook function to define how a route should be included in the sitemap.
+This hook function will be called during build-time only and will be given callbacks to inform the decisions. It has the following signature:
+
+```ts
+type ConfigHook = (callbacks: {
+  addToSitemap: (routeParams?: Record<string, string | undefined>[]) => void;
+  removeFromSitemap: (routeParams?: Record<string, string | undefined>[]) => void;
+  setSitemap: (
+    routeParams: Array<{ sitemap?: boolean; params: Record<string, string | undefined> }>
+  ) => void;
+}) => Promise<void> | void;
+```
+
+As noted by the type, the given hook function can be async, which means you can load remote information.
+
+```astro
+---
+import sitemap from 'sitemap-ext:config';
+
+sitemap(async ({ addToSitemap, removeFromSitemap }) => {
+  const shouldBeIncluded = await fetchThisConfigFromRemote();
+
+  if (shouldBeIncluded) {
+    addToSitemap();
+  } else {
+    removeFromSitemap();
+  }
+});
+---
+```
+
+Not calling any of the given configuration callbacks means the route will follow what is defined by the global [`includeByDefault` option](#includebydefault)
+
+#### `addToSitemap` / `removeFromSitemap`
+
+Calling these configuration callbacks without any argument has the same meaning as the [full opt in/out](#full-opt-inout) using the boolean values.
+
+These functions also allow passing a list of parameter maps in the argument, these parameter maps are the same as the [`params` field in Astro's `getStaticPaths`](https://docs.astro.build/en/reference/api-reference/#params).
+
+On a static route this argument is ignored. On a dynamic route, this argument is used to replace the dynamic segments and add the resulting URLs to the sitemap. This means you can add some pages generated by an Astro route to the sitemap while keeping others out of it.
+
+```astro title="src/pages/[locale]/[postId].astro"
+---
+import { getCollection } from 'astro:content';
+import sitemap from 'sitemap-ext:config';
+
+sitemap(async ({ addToSitemap }) => {
+  const blogPosts = await getCollection('posts');
+
+  addToSitemap(
+    blogPosts.map((post) => ({
+      locale: post.data.locale,
+      postId: post.data.id,
+    }))
+  );
+});
+---
+```
+
+#### `setSitemap`
+
+This function receives an array of parameter maps with a boolean to indicate whether each of those parameters describe a route that should be included or excluded from the sitemap.
```

Comment:
You doN't seem to describe what happens if the boolean is omitted. It seems to be optional based on the type definition.

Reply:
Nice catch